mesop/components/autocomplete/autocomplete.ts

```diff
@@ -0,0 +1,168 @@
+import {MatFormFieldModule} from '@angular/material/form-field';
+import {MatInputModule} from '@angular/material/input';
+import {Component, Input, OnInit} from '@angular/core';
+import {
+  Key,
+  Type,
+  UserEvent,
+  Style,
+} from 'mesop/mesop/protos/ui_jspb_proto_pb/mesop/protos/ui_pb';
+import {
+  AutocompleteType,
+  AutocompleteOptionSet,
+  AutocompleteOptionGroup,
+} from 'mesop/mesop/components/autocomplete/autocomplete_jspb_proto_pb/mesop/components/autocomplete/autocomplete_pb';
+import {Channel} from '../../web/src/services/channel';
+import {formatStyle} from '../../web/src/utils/styles';
+import {Subject} from 'rxjs';
+import {debounceTime, startWith} from 'rxjs/operators';
+import {AsyncPipe, CommonModule} from '@angular/common';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {FormControl, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {
+  MatAutocompleteModule,
+  MatAutocompleteSelectedEvent,
+} from '@angular/material/autocomplete';
+
+@Component({
+  selector: 'mesop-autocomplete',
+  templateUrl: 'autocomplete.ng.html',
+  standalone: true,
+  imports: [
+    MatInputModule,
+    MatFormFieldModule,
+    CommonModule,
+    MatAutocompleteModule,
+    ReactiveFormsModule,
+    AsyncPipe,
+  ],
+})
+export class AutocompleteComponent implements OnInit {
+  @Input({required: true}) type!: Type;
+  @Input() key!: Key;
+  @Input() style!: Style;
+  private _config!: AutocompleteType;
+  filteredOptions: Observable<AutocompleteOptionSet[]> = new Observable<
+    AutocompleteOptionSet[]
+  >();
+  private inputSubject = new Subject<Event>();
+  autocompleteControl = new FormControl('');
+  constructor(private readonly channel: Channel) {
+    this.inputSubject
+      .pipe(
+        // Setting this to a short duration to avoid having the user trigger another event
+        // during this debounce time period:
+        // https://github.com/google/mesop/issues/171
+        debounceTime(150),
```

Comment:
no action-needed: I've been thinking about tweaking the input debounce time higher, but I think it's fine to keep these consistent for now.